framework/src/modules/random/utils.ts

```diff
@@ -94,28 +94,24 @@ export const getRandomSeed = (
 		throw new Error('Height is in the future.');
 	}
 
-	for (const seedObject of currentSeeds) {
-		if (seedObject.valid) {
-			randomSeed = bitwiseXOR([randomSeed, seedObject.seedReveal]);
-		}
-	}
-
-	return randomSeed;
+	return bitwiseXOR(currentSeeds);
 };
 
 export const bitwiseXOR = (bufferArray: Buffer[]): Buffer => {
 	if (bufferArray.length === 1) {
 		return bufferArray[0];
 	}
 
-	const bufferSizes = new Set(bufferArray.map(buffer => buffer.length));
-	if (bufferSizes.size > 1) {
-		throw new Error('All input for XOR should be same size');
+	const size = bufferArray[0].length;
```

Comment:
if we use index, it would be better to check the length of `bufferArray` . we can just throw if the length is 0 at the beginning

Reply:
Added the check for empty array, and the related unit test.